src/main/java/redis/embedded/RealRedisCluster.java

```diff
@@ -0,0 +1,281 @@
+                jedis.clusterAddSlots(master.getSlotRange().getRange());
+                //get node id
+                String curNodeId = getNodeIdFromClusterNodesAck(jedis.clusterNodes());
+                System.out.println(String.format("Master node: %s with slots[%d,%d]",
+                        curNodeId,
+                        master.getSlotRange().first,
+                        master.getSlotRange().last));
+                master.setNodeId(curNodeId);
+            } finally {
+                if (jedis != null) {
+                    jedis.close();
+                }
+            }
+        }
+        return masters;
+    }
+
+    private String getNodeIdFromClusterNodesAck(String ack) {
+        return ack.split(" :")[0];
+    }
+
+    private void setReplicates(List<MasterNode> masters) {
+        for (MasterNode master : masters) {
+            setSlaves(master.getNodeId(), master.getSlaves());
+        }
+    }
+
+    private void setSlaves(String masterNodeId, Set<Redis> slaves) {
+        Jedis jedis = null;
+        for (Redis slave : slaves) {
+            try {
+                //add slots
+                jedis = new Jedis(LOCAL_HOST, slave.ports().get(0));
+                jedis.clusterReplicate(masterNodeId);
+            } finally {
+                if (jedis != null) {
+                    jedis.close();
+                }
+            }
+        }
+    }
+
+    private enum ClusterState {
+        OK("ok"), FAIL("fail");
+        private String state;
+
+        ClusterState(String s) {
+            state = s;
+        }
+
+        public String getState() {
+            return state;
+        }
+
+        public static ClusterState getStateByStr(String s) {
+            for (ClusterState clusterState : ClusterState.values()) {
+                if (s.equals(clusterState.getState())) {
+                    return clusterState;
+                }
+            }
+            throw new IllegalStateException("illegal cluster state: " + s);
+        }
+    }
+
+    private static class MasterNode {
+        final Redis master;
+        String nodeId;
+        final RealRedisCluster.SlotRange slotRange;
+        final Set<Redis> slaves;
+
+        public MasterNode(Redis master, SlotRange slotRange) {
+            this.master = master;
+            this.slotRange = slotRange;
+            slaves = new HashSet<Redis>();
+        }
+
+        public Set<Redis> getSlaves() {
+            return slaves;
+        }
+
+        public void addSlave(Redis slave) {
+            slaves.add(slave);
+        }
+
+        public Redis getMaster() {
+            return master;
+        }
+
+        public RealRedisCluster.SlotRange getSlotRange() {
+            return slotRange;
+        }
+
+        public String getNodeId() {
+            return nodeId;
+        }
+
+        public void setNodeId(String nodeId) {
+            this.nodeId = nodeId;
+        }
+    }
+
+    private static class SlotRange {
```

Comment:
Suggestion, extract all Redis Cluster classes to subpackage: redis.embedded.cluster